internals/src/Tizen.Peripheral/Tizen.Peripheral/Spi.cs

```diff
@@ -0,0 +1,241 @@
+    public class Spi : IDisposable
+    {
+
+        //TODO Provide default values.
+        private SpiMode _transferMode;
+        private BitOrder _bitOrder;
+        private byte _bitsPerWord;
+        private uint _frequency;
+
+        /// <summary>
+        /// Native handle to Spi.
+        /// </summary>
+        private IntPtr _handle;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Opens a SPI slave device.
+        /// </summary>
+        /// <param name="bus">The SPI bus number.</param>
+        /// <param name="chip">The SPI chip select number.</param>
+        public Spi(int bus, int chip)
+        {
+            var ret = NativeSpi.Open(bus, chip, out IntPtr handle);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Closes the SPI slave device.
+        /// </summary>
+        ~Spi()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Closes the SPI slave device.
+        /// </summary>
+        public void Close() => Dispose();
+
+        /// <summary>
+        /// Disposes the Spi.
+        /// </summary>
+        public void Dispose() =>Dispose(true);
+
+        /// <summary>
+        /// Disposes the Spi.
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            NativeSpi.Close(_handle);
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the SPI slave device.
+        /// </summary>
+        /// <param name="size">The size of data buffer.</param>
+        /// <returns>The data buffer.</returns>
+        public byte[] Read(int size)
+        {
+            var length = Convert.ToUInt32(size);
+            byte[] data = new byte[size];
+            var ret = NativeSpi.Read(_handle, data, length);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+
+            return data;
+        }
+
+        /// <summary>
+        /// Writes the bytes data to the SPI slave device.
+        /// </summary>
+        /// <param name="data">The data buffer to write.</param>
+        public void Write(byte[] data)
+        {
+            var length = Convert.ToUInt32(data.Length);
+            var ret = NativeSpi.Write(_handle, data, length);
+            if (ret != Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+        }
+
+        /// <summary>
+        /// Exchanges the bytes data to the SPI slave device.
+        /// </summary>
+        /// <param name="data">The data buffer to write.</param>
+        /// <returns>The data buffer.</returns>
+        public byte[] Transfer(byte[] data)
```

Comment:
Why not following MS API?
``` c#
public void TransferSequential(byte[] writeBuffer, byte[] readBuffer);
```
https://docs.microsoft.com/en-us/uwp/api/windows.devices.spi.spidevice.transfersequential?view=winrt-19041